app/handlers/GuaranteeBalanceResponseHandler.scala

```diff
@@ -34,15 +34,16 @@ import java.time.LocalDateTime
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-sealed trait GuaranteeBalanceResponseHandler extends Logging {
-
-  val sessionRepository: SessionRepository
-  val auditService: AuditService
+class GuaranteeBalanceResponseHandler @Inject() (
+  val sessionRepository: SessionRepository,
+  val auditService: AuditService,
   val errorHandler: ErrorHandler
+)(implicit val ec: ExecutionContext)
+    extends Logging {
 
-  implicit val ec: ExecutionContext
+  private val eoriNumber = "-" // TODO check if we can fetch and use it here to pass audit service
```

Comment:
Very minor but could we remove eoriNumber from the audit model case classes and hard code a hyphen into the detail method

And instead of a TODO could we have a comment saying something along the lines of "EORI is not captured in P5 but remains part of the pre-existing audit structure from P4"